tests/tests/tier0/bluechi-agent-connect-via-manager-address/test_bluechi_agent_connect_via_manager_address.py

```diff
@@ -0,0 +1,66 @@
+# SPDX-License-Identifier: LGPL-2.1-or-later
+
+from typing import Dict
+import os
+import socket
+
+
+from bluechi_test.test import BluechiTest
+from bluechi_test.machine import BluechiControllerMachine, BluechiAgentMachine
+from bluechi_test.config import BluechiControllerConfig, BluechiAgentConfig
+from bluechi_test.util import read_file
+
+NODE_FOO = "node-FOO"
+failed_status = "failed"
+
+
+def exec(ctrl: BluechiControllerMachine, nodes: Dict[str, BluechiAgentMachine]):
+    node_foo = nodes[NODE_FOO]
+    config_file_location = "/var/tmp"
+    bluechi_agent_str = "bluechi-agent"
+    invalid_conf_str = "config/Invalid-ControllerAddress.conf"
+    invalid_controller_address = "Invalid-ControllerAddress.conf"
+    valid_conf_str = "config/valid-ControllerAddress.conf"
+    valid_controller_address = "valid-ControllerAddress.conf"
+    valid_conf_txt = f"[bluechi-agent] \
+    \nControllerAddress=tcp:host={socket.gethostbyname(socket.gethostname())},port=8420"
+    service = "org.eclipse.bluechi"
+    object = "/org/eclipse/bluechi"
+    interface = "org.eclipse.bluechi.Controller"
+    property = "Status"
+
+    content = read_file(invalid_conf_str)
+    node_foo.create_file(config_file_location, invalid_conf_str, content)
+
+    node_foo.restart_with_config_file(os.path.join(config_file_location, invalid_controller_address), bluechi_agent_str)
+    assert node_foo.wait_for_unit_state_to_be(bluechi_agent_str, "active", delay=2)
+    assert "s \"down\"" == ctrl.exec_run(f"busctl get-property {service} {object} {interface} {property}")[1]
+
+    content = read_file(valid_conf_str)
+    node_foo.create_file(config_file_location, valid_conf_str, content)
+
+    f = open(valid_conf_str, "a")
+    f.write(valid_conf_txt)
+    f.close()
+
+    content = read_file(valid_conf_str)
+    node_foo.create_file(config_file_location, valid_conf_str, content)
+
+    node_foo.restart_with_config_file(os.path.join(config_file_location, valid_controller_address), bluechi_agent_str)
+    assert node_foo.wait_for_unit_state_to_be(bluechi_agent_str, "active", delay=2)
+    assert "s \"up\"" == ctrl.exec_run(f"busctl get-property {service} {object} {interface} {property}")[1]
+
+
+def test_agent_invalid_configuration(
+        bluechi_test: BluechiTest,
+        bluechi_node_default_config: BluechiAgentConfig, bluechi_ctrl_default_config: BluechiControllerConfig):
+
+    node_good_cfg = bluechi_node_default_config.deep_copy()
+    node_good_cfg.node_name = NODE_FOO
```

Comment:
Here is the place where you can setup `ControllerAddress` for the foo node, so it's using the address by default instead of host/port

```
node_good_cfg.controller_address=f"tcp:host={node_good_cfg.controller_host},port={node_good_cfg.controller_port}"
node_good_cfg.controller_host=""
node_good_cfg.controller_port=""
```

And for clarity I'd rename `node_good_cfg` to `node_foo_cfg` ...